identifiers/validate.go

```diff
@@ -25,14 +25,16 @@ var (
 	//
 	// Rules for domains, defined in RFC 1035, section 2.3.1, are used for
 	// identifiers.
-	identifierRe = regexp.MustCompile(reAnchor(label + reGroup("[.]"+reGroup(label)) + "*"))
+	identifierRe = regexp.MustCompile(reAnchor(reCapture(label) + reGroup("[.]"+reCapture(label)) + "*"))
 
 	errIdentifierInvalid = errors.Errorf("invalid, must match %v", identifierRe)
+
+	errLengthInvalid = errors.Errorf("invalid, name must less than 255, and label must less than 63.")
```

Comment:
Why create another error type and not expose it? Also, this substructure isn't something that should be exposed to the caller.